src/sentry/tasks/unmerge.py

```diff
@@ -470,15 +468,34 @@ def update_tag_value_counts(id_list):
         )
 
 
+def lock_hashes(project_id, source_id, fingerprints):
+    with transaction.atomic():
+        eligible_hashes = list(
```

Comment:
(Specifically, this is to avoid trying to start another migration on a hash that is already being migrated by another process.)